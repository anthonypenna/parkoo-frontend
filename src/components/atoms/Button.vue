<template>
  <button v-bind="$attrs" :class="`button button--${theme}`">
    <slot />
  </button>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  name: 'Button',

  props: {
    theme: {
      default: 'secondary',
      type: String
    }
  }
})
</script>

<style lang="scss" scoped>
.button {
  max-width: 295px;
  width: 100%;
  padding: 14px 0;
  border: none;
  border-radius: 5px;
  font-family: inherit;
  font-size: 16px;
  font-weight: bold;

  &--primary {
    background: #ff9979;
    color: #fff;
  }

  &--secondary {
    background: #f1f1f1;
    color: #000;
  }
}
</style>
