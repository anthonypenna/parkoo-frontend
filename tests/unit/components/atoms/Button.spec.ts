import { shallowMount } from '@vue/test-utils'
import Button from '@/components/atoms/Button.vue'

describe('<Button />', () => {
  describe('with primary theme', () => {
    it('should render correctly', () => {
      const theme = 'primary'
      const wrapper = shallowMount(Button, { propsData: { theme } })
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('with secondary theme', () => {
    it('should render correctly', () => {
      const theme = 'secondary'
      const wrapper = shallowMount(Button, { propsData: { theme } })
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  it('should render the correct slot', () => {
    const wrapper = shallowMount(Button, { slots: { default: 'Hello world!' } })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
