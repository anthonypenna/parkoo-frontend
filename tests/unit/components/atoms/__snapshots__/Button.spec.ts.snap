// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the correct slot 1`] = `<button class="button button--secondary">Hello world!</button>`;

exports[`<Button /> with primary theme should render correctly 1`] = `<button class="button button--primary"></button>`;

exports[`<Button /> with secondary theme should render correctly 1`] = `<button class="button button--secondary"></button>`;
